software/cm/ansible/src/main/java/org/apache/brooklyn/entity/cm/ansible/AnsiblePlaybookTasks.java

```diff
@@ -105,5 +107,27 @@ private static String optionalExtraVarsParameter(Object extraVars) {
         }
         return " --extra-vars \"@" + EXTRA_VARS_FILENAME + "\" ";
     }
+
+    public static TaskFactory<?> setUpHostsFile(String installDir, boolean force) {
+        final String hostsFile = "/etc/ansible/hosts";
+        final String tempFile = "/tmp/hosts";
+        String checkInstalled = !force ? "which ansible || " : "";
+        return sshCommands(
+                checkInstalled + " {",
+                sudo("rm -f " + hostsFile),
+                "echo 'localhost ansible_connection=local' > " + tempFile,
```

Comment:
I needed to write it as sudo, but when you try to do the create directly under sudo with a redirect, it breaks (redirects don't work under sudo).  The operation is still atomic really, it is the "mv"  - the tempFile is just a convenience.